code/controllers/UserController.php

```diff
@@ -66,27 +79,35 @@ public function sendInvite($data, Form $form)
         }
 
         $invite = UserInvitation::create();
-        $form->saveInto($invite);
-        try {
-            $invite->write();
-        } catch (ValidationException $e) {
+        if (!Permission::check('ACCESS_USER_INVITATIONS')){
```

Comment:
Move this check right to the top of the sendInvite method (before $form->validate()) and return early if the permission check fails. 
```php
if(!Permission::check('ACCESS_USER_INVITATIONS'){
    $form->sessionMessage(
        _t('UserController.INVITATION_PERMISSION_DENIED', 'You don\'t have permission to create user 
        invitations'),
        'bad'
    );
    return $this->redirectBack();
}
```